api/posts/posts-router.js

```diff
@@ -1 +1,66 @@
 // implement your posts router here
+const express = require('express')
+const poster = require('./posts-model')
+const router = express.Router();
+
+router.get('/',(req,res)=> {
+    poster.find()
+    .then((posts)=>{
+        res.status(200).json(posts);
+    })
+    .catch((error)=>{
+        res.status(500).json({ message: "The posts information could not be retrieved"});
+    })
+})
+
+router.get('/:id',(req,res)=>{
+    poster.findById(req.params.id)
+    .then((post)=>{
+        if(post){
+            res.status(200).json(post);
+        }else {
+            res.status(404).json({message: "The post with the specified ID does not exist"});
+        }
+    })
+    .catch((error)=>{
+        res.status(500).json({ message: "The post information could not be retrieved" });
+    })
+
+})
+
+
+router.post('/',(req,res)=>{
+    if(!req.body.title || req.body.contents){
+        res.status(400).json({ message: "Please provide title and contents for the post" });
```

Comment:
trebuie sa ii faci return ca daca nu pai atunci o sa mergi mai departe si o sa incerci sa creezi in baza de date